packages/slate-editor/src/extensions/image/components/ImageElement.tsx

```diff
@@ -100,9 +100,12 @@ export function ImageElement({
         [withSizeOptions, element.file],
     );
 
+    const { ref, ...slateAttributes } = attributes;
+
     return (
         <ResizableEditorBlock
-            {...attributes}
+            {...slateAttributes}
+            ref={isSupportingCaptions ? undefined : ref}
```

Comment:
Nope. The given `ref` is required to be set on the top-level DOM node for the element. Otherwise a lot of internal logic inside Slate just breaks.

![image](https://user-images.githubusercontent.com/370680/197748494-91520ee3-bb7b-48f1-b2ce-d1e84fbb86c4.png)

Reply:
Do you have some other ideas how that can be fixed?

Reply:
I'd try to look into two things that might be causing this, or might provide a way to fix this:

- Our homebrew `useCursorInView` hook
- Slate native `scrollSelectionIntoView` property

Reply:
Looks like `useCursorInView `. Thanks for pointing the direction ❤️